portal-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.js

```diff
@@ -0,0 +1,82 @@
+const api = require('../../api');
+
+const getExistingYear = (groupedReports, year) => {
+  let existingYear;
+  for (let i = 0; i < groupedReports.length; i += 1) {
+    if (groupedReports[i].year === year) {
+      existingYear = groupedReports[i];
+      break;
+    }
+  }
+
+  return existingYear;
+};
+
+const getMonthName = (monthNumber) => {
+  // date is set to 1st Jan to avoid bug when today's month has more days than target month
+  const date = new Date(2023, 1, 1);
+  // offset by 1 as January = 1 in Database
+  date.setMonth(monthNumber - 1);
+  return date.toLocaleString('default', { month: 'long' });
+};
+
+const getGroupedReports = (reports) => {
+  const groupedReports = [];
+  if (reports.length) {
+    reports.forEach((report) => {
+      const { year } = report;
+      const existingYear = getExistingYear(groupedReports, year);
+
+      if (existingYear) {
+        existingYear.reports.push({
+          month: getMonthName(report.month),
+          path: report.path,
+        });
+      } else {
+        if (groupedReports.length) {
+          for (let i = 1; i < year - groupedReports[groupedReports.length - 1].year; i += 1) {
+            const checkExistingYear = getExistingYear(groupedReports, year - i);
+            if (!checkExistingYear) {
+              groupedReports.push({
+                year: year - i,
+                reports: [],
+              });
+            }
+          }
+        }
+
+        groupedReports.push({
+          year,
+          reports: [{
+            month: getMonthName(report.month),
+            path: report.path,
+          }]
+        });
+      }
+    });
+  }
+
+  return groupedReports;
+};
+
+const getPreviousReportsByBankId = async (req, res) => {
+  try {
+    const { bankId } = req.params;
+
+    // Add validation for user bank ID here?
```

Comment:
Can see you've added a comment here so ye just chuck it in a route validation and then in the api files